### PR TITLE
Feature/carousel

### PR DESCRIPTION
**Pull request by ydomenjoud**

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
+import { Directive, ElementRef, EmbeddedViewRef, Host, Input, OnInit, Optional, TemplateRef, ViewContainerRef } from '@angular/core';
+import { KalCarouselComponent, KalCarouselItemStatus } from './kal-carousel.component';
+import { animate, AnimationBuilder, AnimationFactory, AnimationPlayer, style } from '@angular/animations';
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts:13]: imports

**Review by jsp-up — DISMISSED** [review]

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  private player: AnimationPlayer;
+
+  private animationRunning = false;
+
+  constructor(
+    private viewContainerRef: ViewContainerRef,
+    private template: TemplateRef<CarouselContext<T>>,
+    private builder: AnimationBuilder,
+    private elementRef: ElementRef,
+    @Host() @Optional() private carousel: KalCarouselComponent<T>) {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts]: you can use fordwardRef

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  }
+
+  get currentItem() {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts]: Type T

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    return this.carousel.currentItem;
+  }
+
+  get items() {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts]: Type T

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    return this.carousel.items;
+  }
+
+  private get length() {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts]: Type method

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+  private render(status: KalCarouselItemStatus) {
+    switch (status) {
+      case KalCarouselItemStatus.Init:
+        this.items.forEach((item: T, index: number) => {
+          const context = this.editContext(item, index, status);
+          this.viewContainerRef.createEmbeddedView(this.template, context);
+        });
+        break;
+      case KalCarouselItemStatus.Append:
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-carousel/kal-carousel-item.directive.ts:77]: default

File: src/app/atoms/carousel/carousel.component.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  elements = [];
+
+  vignetteVisible  = 4;
+
+  constructor() {
+    for (let i = 1; i < 6; i++) {
+      this.elements.push(
+        {
+          title: 'image ' + i,
+          image: 'http://6tech.net/img/test/' + i + '.jpg'
```
Comment (MichaelGrosio) [inline src/app/atoms/carousel/carousel.component.ts]: img

**Review by MichaelGrosio — CHANGES_REQUESTED** [review]

**Review by jsp-up — APPROVED** [review]

**Review by MichaelGrosio — APPROVED** [review]

**Pull request merged**